src/gui/shell.cpp

```diff
@@ -161,12 +143,96 @@ bool Shell::setGuiFont(const QString& fdesc, bool force, bool updateOption, bool
 	// Font changed: trigger resize to update the ShellWidget, and update font variables.
 	resizeNeovim(size());
 
-	m_nvim->api0()->vim_set_var("GuiFont", fontDesc());
+	m_nvim->api0()->vim_set_var("GuiFont", fontDesc(false));
 
 	// Updating guifont when the user has already called 'set guifont=...' may cause
 	// unwanted recursion. Only update this option for ':GuiFont', and dialog calls.
 	if (isGuiDialogRequest || updateOption) {
-		m_nvim->api0()->vim_set_option("guifont", fontDesc());
+		m_nvim->api0()->vim_set_option("guifont", fontDesc(false));
+	}
+
+	return true;
+}
+
+bool Shell::setGuiFontWide(const QString& fdesc, bool force, bool updateOption)
+{
+	bool setShellFontSuccess{ false };
+	const bool isGuiDialogRequest{ fdesc == "*" };
+
+	if (isGuiDialogRequest) {
+		bool fontDialogSuccess = false;
+		QFont font{ QFontDialog::getFont(&fontDialogSuccess, QWidget::font(), this, {},
+			QFontDialog::MonospacedFonts) };
+
+		// The font selection dialog was canceled by the user.
+		if (!fontDialogSuccess) {
+			return false;
+		}
+
+		setShellFontSuccess = setShellFont(font.family(), font.pointSize(), font.weight(),
+			font.italic(), force, true);
+	}
+	else {
+		const QStringList wideFontList{ fdesc.split(',') };
+		for (const auto& wfdesc : wideFontList)
+		{
+			// Exit early if the font description does not change, prevent loops.
+			if (wfdesc.compare(fontDesc(true), Qt::CaseInsensitive) == 0) {
```

Comment:
This should be outside of the for loop.

This checks if the user has changed the value of the font register. IE there is no need to do anything when the register remains unchanged.